repos/dir.targets

```diff
@@ -185,6 +186,24 @@
                        Condition="'@(_BuiltPackages)'!=''" />
   </Target>
 
+  <Target Name="ExtractToolPackage"
+          DependsOnTargets="GatherBuiltPackages"
+          AfterTargets="Build"
+          Condition="'$(BuiltToolPackageId)' != ''">
+    <ItemGroup>
+      <_AllSourceBuiltPackages Include="$(SourceBuiltPackagesPath)*.nupkg" />
+      <_ToolPackage Include="@(_AllSourceBuiltPackages)"
+                    Condition="$([System.String]::new('%(Filename)').StartsWith('$(BuiltToolPackageId)'))" />
+    </ItemGroup>
```

Comment:
I'm not too happy with this target. I think ideally we'd have something that would `dotnet restore` the toolset package based on `PackageVersions.props` to avoid the `StartsWith` fuzziness and manual zip file extraction.